src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -362,6 +362,8 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			const processContext = ProcessExecutionSupportedContext.bindTo(this._contextKeyService);
 			processContext.set(process && !isVirtual);
 		}
+		// update tasks so an incomplete list isn't returned when getWorkspaceTasks is called
+		this._workspaceTasksPromise = undefined;
```

Comment:
I considered calling `updateWorkspaceTasks` here instead of doing this, but then it's possible that not all of the extensions will have been activated by that time and we don't want to set the promise if that's the case either